ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RequestHandler.java

```diff
@@ -102,14 +114,19 @@ public KsqlEntityList execute(
       final ServiceContext serviceContext,
       final ConfiguredStatement<T> configured,
       final Map<String, Object> mutableScopedProperties,
-      final KsqlEntityList entities
+      final KsqlEntityList entities,
+      final ProducerTransactionManager producerTransactionManager
   ) {
     final Class<? extends Statement> statementClass = configured.getStatement().getClass();
     commandQueueSync.waitFor(new KsqlEntityList(entities), statementClass);
 
     final StatementExecutor<T> executor = (StatementExecutor<T>)
         customExecutors.getOrDefault(statementClass, distributor);
 
+    if (executor instanceof DistributingExecutor) {
+      ((DistributingExecutor) executor).setTransactionManager(producerTransactionManager);
```

Comment:
I wasn't sure of the best way to handle passing the producerTransactionManager to DistributingExecutor. The interface for StatementExecutor.execute() is shared among many of the non-distributed executors so I didn't think it was appropriate to modify it to pass along the manager.

Reply:
I think customExecutors never contains a DistributingExecutor, so if you don't find the executor in customExecutors you could just create a new DistributingExecutor and you can pass the producerTransactionManager into the constructor of that, so you wouldn't need a setTransactionManager method.

Reply:
There already exists a DistributingExecutor which is set as the executor for the current statement if the statementClass isn't found in the customExecutor.

https://github.com/confluentinc/ksql/blob/master/ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RequestHandler.java#L47

Reply:
+1 to @purplefox's suggestion. We just need to rejigger the code a bit. `DistributingExecutor` doesn't need to implement `StatementExecutor`, and it's `execute` method can take a `TransactionalProducer` as a parameter. Then, `RequestHandler` can just instantiate a wrapper object that _does_ implement `StatementExecutor`, or just use a lambda:

```
final StatementExecutor<T> executor = customExecutors.getOrDefault(
    statementClass,
    (stmt, props, ctx, svcCtx -> distributor.execute(stmt, props, ctx, svcCtx, transactionalProducer));
```

Reply:
^ that works, I'll use that then.